schema.json

```diff
@@ -117,7 +119,7 @@
             "additionalProperties": {
                 "anyOf": [
                     { "$ref": "#" },
-                    { "$ref": "#/definitions/stringArray" }
+                    { "$ref": "#/definitions/nonEmptyStringArray" }
```

Comment:
We could also consider if both `dependencies` and `required` should allow any number of items.

They effectively do the same thing, after all.